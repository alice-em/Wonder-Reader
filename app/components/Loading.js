import CircularProgress from '@material-ui/core/CircularProgress';
import Paper from '@material-ui/core/Paper';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '@material-ui/core/styles/withStyles';

const styles = theme => ({
  LoaderElement: {
    position: 'absolute',
    top: '0',
    left: '0',
    height: '100vh',
    width: '100vw',
    zIndex: '1301',
    backgroundColor: 'rgba(0,0,0,0.2)'
  },
  Paper: {
    height: '50px',
    width: '50px',
    margin: 'auto',
    marginTop: '30vh'
  },
  root: theme.mixins.gutters({
    flexGrow: 1,
    paddingTop: 16,
    paddingBottom: 16,
    marginTop: theme.spacing.unit * 3
  })
});

const PaperElement = ({ classes }) => (
  <Paper className={classes.root} elevation={4} style={styles.Paper}>
    <CircularProgress
      className={classes.progress}
      color="secondary"
      size={50}
    />
  </Paper>
);

const LoaderElement = ({ classes }) => (
  <div style={styles.LoaderElement}>
    <PaperElement classes={classes} />
  </div>
);

const Loading = ({ classes, isLoading }) =>
  isLoading ? <LoaderElement classes={classes} /> : null;

Loading.propTypes = {
  classes: PropTypes.object.isRequired, // eslint-disable-line
  isLoading: PropTypes.bool.isRequired
};

export default withStyles(styles)(Loading);
