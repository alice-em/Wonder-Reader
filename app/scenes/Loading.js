import React from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';

const styles = theme => ({
  LoaderElement: {
    position: 'absolute',
    top: '0',
    left: '0',
    height: '100vh',
    width: '100vw',
    zIndex: '1301',
    backgroundColor: 'rgba(0,0,0,0.2)'
  },
  Paper: {
    height: '50px',
    width: '50px',
    margin: 'auto',
    marginTop: '30vh'
  },
  root: theme.mixins.gutters({
    flexGrow: 1,
    paddingTop: 16,
    paddingBottom: 16,
    marginTop: theme.spacing.unit * 3
  })
});

const Loading = ({
  classes,
  loading
}) => (loading ? <LoaderElement classes={classes} /> : null);

Loading.propTypes = {
  classes: PropTypes.object.isRequired,
  loading: PropTypes.bool.isRequired
};

const LoaderElement = ({ classes }) => (
  <div style={styles.LoaderElement}>
    <PaperElement classes={classes} />
  </div>
);

LoaderElement.propTypes = {
  classes: PropTypes.object.isRequired
};

const PaperElement = ({ classes }) => (
  <Paper className={classes.root} elevation={4} style={styles.Paper}>
    <CircularProgress
      className={classes.progress}
      color="secondary"
      size={50}
    />
  </Paper>
);

PaperElement.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(Loading);
