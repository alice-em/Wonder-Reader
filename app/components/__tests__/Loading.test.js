import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';

import { Loading, PaperElement } from '../Loading';

Enzyme.configure({ adapter: new Adapter() });

describe('Loading', () => {
  const props = {
    classes: {
      root: 'sampleRoot',
      progress: 'sampleProgress',
    },
    isLoading: true,
  };
  it('should render', () => {
    const wrapper = shallow(<Loading {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render PaperElement', () => {
    const wrapper = shallow(<PaperElement {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
