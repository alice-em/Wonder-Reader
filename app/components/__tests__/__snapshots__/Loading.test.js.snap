// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading
    classes={
      Object {
        "progress": "sampleProgress",
        "root": "sampleRoot",
      }
    }
    isLoading={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "progress": "sampleProgress",
        "root": "sampleRoot",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "progress": "sampleProgress",
          "root": "sampleRoot",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Loading should render PaperElement 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PaperElement
    classes={
      Object {
        "progress": "sampleProgress",
        "root": "sampleRoot",
      }
    }
    isLoading={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Paper)
        className="sampleRoot"
        elevation={4}
      >
        <WithStyles(CircularProgress)
          className="sampleProgress"
          color="secondary"
          size={50}
        />
      </WithStyles(Paper)>,
      "style": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(CircularProgress)
          className="sampleProgress"
          color="secondary"
          size={50}
        />,
        "className": "sampleRoot",
        "elevation": 4,
        "style": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "sampleProgress",
          "color": "secondary",
          "size": 50,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Paper)
          className="sampleRoot"
          elevation={4}
        >
          <WithStyles(CircularProgress)
            className="sampleProgress"
            color="secondary"
            size={50}
          />
        </WithStyles(Paper)>,
        "style": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(CircularProgress)
            className="sampleProgress"
            color="secondary"
            size={50}
          />,
          "className": "sampleRoot",
          "elevation": 4,
          "style": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "sampleProgress",
            "color": "secondary",
            "size": 50,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
